lattices/LatticeMath/LatticeStatistics.tcc

```diff
@@ -987,11 +1008,36 @@ void LatticeStatistics<T>::_doStatsLoop(
             (*progressMeter)++;
         }
     }
+    if (! doRobust_p) {
+        // zero out the quantile stats since they will not be computed.
+        // For the old TiledApply method, this is done by zeroing out
+        // the array prior to computing the stats (see above FIXME comment
+        // in the current method that explains that).
+        uInt ndim = pStoreLattice_p->ndim();
+        IPosition arrShape = pStoreLattice_p->shape().removeAxes(
+            IPosition(1, ndim - 1)
+        );
+        Array<AccumType> zeros(arrShape, AccumType(0));
+        IPosition start(ndim, 0);
+        start[ndim - 1] = LatticeStatsBase::MEDABSDEVMED;
+        pStoreLattice_p->putSlice(zeros, start);
+        start[ndim - 1] = LatticeStatsBase::MEDIAN;
+        pStoreLattice_p->putSlice(zeros, start);
+        start[ndim - 1] = LatticeStatsBase::Q1;
+        pStoreLattice_p->putSlice(zeros, start);
+        start[ndim - 1] = LatticeStatsBase::Q3;
+        pStoreLattice_p->putSlice(zeros, start);
+        start[ndim - 1] = LatticeStatsBase::QUARTILE;
+        pStoreLattice_p->putSlice(zeros, start);
+    }
 }
 
 template <class T>
 void LatticeStatistics<T>::generateRobust () {
-    Bool showMsg = haveLogger_p && doRobust_p && displayAxes_p.nelements()==0;
+    if (! doRobust_p) {
```

Comment:
might be nicer to just not call the function when its not needed instead of having some extra state here

the function used to just zero out stuff when this was false before?